restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java

```diff
@@ -173,6 +176,73 @@ public Response postDoc(
         });
   }
 
+  @POST
+  @ManagedAsync
+  @ApiOperation(
+      value = "Write multiple documents in one request",
+      notes =
+          "Auto-generates an ID for the newly created document if an idPath is not provided as a query parameter. When an idPath is provided, this operation is idempotent.",
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 201, message = "Created", response = WriteDocResponse.class),
+        @ApiResponse(code = 400, message = "Bad request", response = Error.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 403, message = "Forbidden", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
+      })
+  @Path("collections/{collection-id}/batch")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response writeManyDocs(
+      @Context HttpHeaders headers,
+      @Context UriInfo ui,
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String authToken,
+      @ApiParam(value = "the namespace that the collection is in", required = true)
+          @PathParam("namespace-id")
+          String namespace,
+      @ApiParam(value = "the name of the collection", required = true) @PathParam("collection-id")
+          String collection,
+      @ApiParam(
+              value = "A JSON Lines payload where each line is a document to write",
+              required = true)
+          @NonNull
+          InputStream payload,
+      @ApiParam(
+              value =
+                  "The path where an ID could be found in each document. If defined, the value at this path will be used as the ID for each document. Otherwise, a random UUID will be given for each document.",
+              required = false)
+          @QueryParam("id-path")
+          String idPath,
+      @Context HttpServletRequest request) {
+    // This route does nearly the same thing as PUT, except that it assigns an ID for the requester
+    // And returns it as a Location header/in JSON body
+    logger.debug("Batch Write: Collection = {}", collection);
+    return handle(
+        () -> {
+          List<String> idsCreated =
+              documentService.writeManyDocs(
+                  authToken,
+                  namespace,
+                  collection,
+                  payload,
+                  Optional.ofNullable(idPath),
+                  dbFactory,
+                  getAllHeaders(request));
+
+          return Response.created(
```

Comment:
WDYT about `202` (Accepted) instead? Currently our code does not guarantee that anything was created at this point or will eventually be created (see my other comment about exceptions).

Even if we wait for the ack from storage, batches may or may not be applied by the time we get the ack, so `201` (Created) might be an overstatement. WDYT?

Reply:
+1 for `202` since we're just "accepting" here

Reply:
Ok, let's switch to `202` :)